File: 206-reverse-linked-list/206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head,ListNode *prev = NULL) {
        if(head == NULL)
            return head;
        ListNode *p = reverseList(head->next,head);
        if(p == NULL)
            p=head;
        head->next = prev;
        return p;
    }
};

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head,prev = null) {
    if(head === null)
        return head;
    let  p = reverseList(head.next,head);
    if(p === null)
        p = head;
    head.next = prev;
    return p;
};